### PR TITLE
RFC: only warn if some package couldn't update

### PR DESCRIPTION
**Pull request by KristofferC**
Change this:

image

into something like:

image

### CODE REVIEW

File: base/pkg/entry.jl [hunk 1]
```diff
@@ -380,15 +380,14 @@ function update(branch::AbstractString, upkgs::Set{String})
...
+    deferred_warnings = String[]
     avail = Read.available()
     # this has to happen before computing free/fixed
     for pkg in filter(Read.isinstalled, collect(keys(avail)))
         try
             Cache.prefetch(pkg, Read.url(pkg), [a.sha1 for (v,a)=avail[pkg]])
         catch err
-            cex = CapturedException(err, catch_backtrace())
-            push!(deferred_errors, PkgError("Package $pkg: unable to update cache.", cex))
+            push!(deferred_warnings, string("Package $pkg: unable to update cache\n    └ ", err, "\n"))
```
Comment (ararslan) [inline base/pkg/entry.jl]: Is that character commonly available in terminals? IIRC the Windows terminal has some issues with certain box drawing characters.

**Comment by ararslan** [comment]
Overall 💯, I think this is a huge improvement!

**Comment by tkelman** [comment]
The update operation failed, it's an error and should be shown as such. We can add mitigations for this specific cause, but wholesale making all potential errors non-fatal is the wrong approach here.

**Comment by KristofferC** [comment]
It depends what you interpret `Pkg.update` to do. To me, it is: update the packages you can, which is done successfully. It is the same as not throwing an error because a package is dirty and that package thus fails to update. 

If it is a user error to havel local packages this should he documented to say it is not supported.

**Comment by tkelman** [comment]
We should do something better and more specific about this particular issue of not having tracking information for a branch. Not unconditionally start treating anything else that might go wrong as less of a problem, that would hide actual serious issues.

**Comment by KristofferC** [comment]
Ok, so do what I did but only in the case of no tracking information?

**Comment by tkelman** [comment]
Yes, this particular issue about not having tracking information should probably be a warning. But the change here is not a good way of going about that, you're making any possible failure, including major bugs in the libgit2 implementation or who knows what else might otherwise throw an error, into non-fatal warnings. Addressing a single specific issue shouldn't be done by turning absolutely all errors into warnings. Address the known issue in a targeted way, as specific as possible to avoid making anything else unexpected or unintended non-fatal (if we don't know exactly what caused it in advance, it should remain an error).

**Comment by StefanKarpinski** [comment]
Have to agree – having unexpected errors that get muffled leads to lots more pain down the road.

**Comment by KristofferC** [comment]
No error messages got "muffled" here, they just got changed to warnings. I'm not really into how the libgit2 versions work so I don't think I'm equipped to do this in the right way. I'll just run with this locally for now to avoid the hundreds of lines of backtraces I get at every Pkg.update()

**Pull request closed without merging**